### PR TITLE
Added new network configuration proposal (fate#326480)

### PR DESCRIPTION
**Pull request by teclator**
- [Trello Card](https://trello.com/c/DLi2b1mG/338-fate-326480-add-network-to-installation-proposal-overview-page)

## Added network proposal
networkproposal

## Compact the current information

completenetworkproposal

### CODE REVIEW

File: src/lib/network/lan_items_summary.rb [hunk 1]
```diff
@@ -109,5 +132,61 @@ def ifcfg_protocol(ifcfg)
         LanItems.DeviceProtocol(ifcfg)
       end
     end
+
+    # Return the label name for the given interface
+    #
+    # @return [String] label name for the given interfave
+    def summary_label_for(name)
```
Comment (mchf) [inline src/lib/network/lan_items_summary.rb]: missing ```@param```

File: src/lib/network/lan_items_summary.rb [hunk 2]
```diff
@@ -109,5 +132,61 @@ def ifcfg_protocol(ifcfg)
...
+    end
+
+    # Convenience method that obtains the list of static configured interfaces
+    #
+    # @return [Array<String>] static configured interface names
+    def static_ifaces
+      LanItems.find_static_ifaces
+    end
+
+    # Convenience method that obtains the list of static configured interfaces
```
Comment (mchf) [inline src/lib/network/lan_items_summary.rb]: no ... all devices which have an ifcfg file

File: src/lib/network/lan_items_summary.rb [hunk 3]
```diff
@@ -109,5 +132,61 @@ def ifcfg_protocol(ifcfg)
...
+    #
+    # @return [Array<String>] static configured interface names
+    def configured_ifaces
```
Comment (mchf) [inline src/lib/network/lan_items_summary.rb]: seems unused

File: src/lib/network/lan_items_summary.rb [hunk 4]
```diff
@@ -109,5 +132,61 @@ def ifcfg_protocol(ifcfg)
...
+
+    # Return the label name for the given interface
+    #
+    # @return [String] label name for the given interfave
+    def summary_label_for(name)
+      case NetworkInterfaces.GetType(name)
+      when "br"
+        "#{name} (Bridge)"
+      when "bond"
+        "#{name} (Bond Master)"
```
Comment (mchf) [inline src/lib/network/lan_items_summary.rb]: I'm not sure here but ```_()``` may be needed

**Comment by mchf** [comment]
I was thinking of it one more time and I'd go for even more minimalist solution. I mean something like:
```
- Configured with DHCP: eth1, eth2, ...
- Statically configured: eth0, ...
- Bridges: br0, ...
...
```
for quick overview this is enough in my POV (it answers questions like: Did I forget to configure eth0 or not?) all other details are available when the caption is clicked - why to show static ip in the overview when dynamic one is not shown etc. So, in short I'd make the overview even more compact and high level.

also please notice that there are no empty lines between bullets in other proposals ... it should look consistently.

**Comment by teclator** [comment]
Already realized the extra line between bullets, I thought it was intentional for device list entries and was not sure if remove or not, so for me is completely fine to remove them.

Also fine with a more compact overview, the idea about having the bridges and bondings apart was because the enslaved interfaces (just to be reflected in some section). 

So the main doubt at this point is if we should also collect all of them in the same entry, or maybe just having something like:

`
- Slaves interfaces: eth0, eth1 (Bridge br0) - eth2, eth3 (Bonding bond0)
`

**Comment by mchf** [comment]
Well it depends, I'd prefer:
```
- Bridges: br0 (eth0, eth1), br1 (bond0, bond1)
- Bondings: bond0 (eth4, eth5)
```
bcs you can e.g. have bridge on top of bonded devices and bond device / bridge is more interesting device for me then enslaved ones.

**Comment by mchf** [comment]
and I don't know if the empty line was intentional or not but now is new era ... so remove it ;-)

**Comment by teclator** [comment]
@mchf I buy it ;)

**Comment by teclator** [comment]
Yep, I also realized that, and probably we should use `Bonds` or `Bonding interfaces`, `Bondings` is not very common.

**Comment by teclator** [comment]
@mchf sorted

**Comment by mchf** [comment]
Well I usually use bond device(s) ... You're right that "Bondings" looks strange ;-)

Otherwise LGTM

**Pull request merged**